### PR TITLE
feat(DWS): add a new resource to manage external datasource

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDwsExtDataSource_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws -v

=== RUN   TestAccDwsExtDataSource_basic
=== PAUSE TestAccDwsExtDataSource_basic
=== CONT  TestAccDwsExtDataSource_basic
--- PASS: TestAccDwsExtDataSource_basic (1535.48s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws	1535.523s
```

### CODE REVIEW

File: docs/resources/dws_ext_data_source.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) The DWS cluster ID to which the external data source belongs.
+  The cluster **type** must is **ANALYSIS**.
```
Comment (ShiChangkuo) [inline docs/resources/dws_ext_data_source.md]:
```suggestion
  The cluster **type** must be **ANALYSIS**.
```
is to be

File: docs/resources/dws_ext_data_source.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  The name can contain 1 to 64 characters. Only letters, digits, and underscores (_) are allowed.
+
+  Changing this parameter will create a new resource.
+
+* `type` - (Required, String, ForceNew) The type of the external data source.  
+  The valid values are **OBS**, and **MRS**.
+
+  -> The OBS data source is supported only in version 8.2.0 or later of cluster.
+
+  Changing this parameter will create a new resource.
```
Comment (ShiChangkuo) [inline docs/resources/dws_ext_data_source.md]: it's better to exchange the order of the two lines.

File: huaweicloud/services/dws/resource_huaweicloud_dws_ext_data_source.go [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+	updateDwsExtDataSourceChanges := []string{
+		"reboot",
+		"user_name",
+	}
+
+	if d.HasChanges(updateDwsExtDataSourceChanges...) {
+		// updateDwsExtDataSource: update the DWS external data source.
+		var (
+			updateDwsExtDataSourceHttpUrl = "v1.0/{project_id}/clusters/{cluster_id}/ext-data-sources/{id}"
+			updateDwsExtDataSourceProduct = "aws"
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_ext_data_source.go]: please check the product name is aws or dws?

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_ext_data_source_test.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+					resource.TestCheckResourceAttrSet(rName, "created_at"),
+					resource.TestCheckResourceAttrSet(rName, "updated_at"),
+				),
+			},
+			{
+				ResourceName:            rName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateIdFunc:       testDwsExtDataSourceImportState(rName),
+				ImportStateVerifyIgnore: []string{"user_pwd", "reboot"},
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_ext_data_source_test.go:104]: please add the ignored fields in docs.

File: docs/resources/dws_ext_data_source.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/dws_ext_data_source.md]: done

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_ext_data_source_test.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_ext_data_source_test.go:104]: ok,done

File: huaweicloud/services/dws/resource_huaweicloud_dws_ext_data_source.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_ext_data_source.go]: it's dws

File: docs/resources/dws_ext_data_source.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/dws_ext_data_source.md]: done

**Pull request merged**